### PR TITLE
refactor(api): Replaced OTP code from alphanumeric to numeric

### LINKED ISSUE #211: API: Change OTP code from alphanumeric to numeric
## Description

Currently, the API sends an OTP while signing in or signing up with an email address. The problem isl, as per the standards, we should be sending a numeric OTP. But currently, it is alphanumeric. We would like to make this change.

## Solution

- Change the [OTP generation method](https://github.com/keyshade-xyz/keyshade/blob/develop/apps/api/src/auth/service/auth.service.ts#L47) to generate a 6-digit numeric code.

## Additional context

Make sure the tests pass. You won't need to add any other tests for this issue.

### PR DESCRIPTION
**Pull request by Ritika1705**
## Description

Replaced RandomUUid with Math.Random <br />

Fixes #211 

## Dependencies

_Mention any dependencies/packages used_

## Future Improvements

_Mention any improvements to be done in future related to any file/feature_

## Mentions

_Mention and tag the people_

## Screenshots of relevant screens

_Add screenshots of relevant screens_

## Developer's checklist

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-check on my work

**If changes are made in the code:**

- [ ] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [ ] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [ ] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [ ] I have made the necessary updates to the documentation, or no documentation changes are required.

___

### **PR Type**
enhancement

___

### **Description**
- Updated OTP generation method in `auth.service.ts` to generate numeric codes using `Math.random()` instead of alphanumeric codes with `randomUUID()`.
- This change affects both the creation and updating of OTP codes.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>auth.service.ts</strong><dd><code>Change OTP Generation to Use Numeric Codes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/api/src/auth/service/auth.service.ts
<li>Changed OTP generation from using <code>randomUUID()</code> to <code>Math.random()</code>.<br> <li> OTP codes are now purely numeric instead of alphanumeric.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/230/files#diff-b133cdd3d1772bc73698061218512fd024ce1ed7cd9eaeb27858e6acb444b90b">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by rajdip-b** [comment]
Hey @Ritika1705, thanks for making this PR! I will check this ASAP and try to merge it.

File: apps/api/src/auth/service/auth.service.ts
Comment (rajdip-b) [inline apps/api/src/auth/service/auth.service.ts:1]: `import { randomUUID } from 'crypto'`

Can you remove this import?

**Review by rajdip-b — CHANGES_REQUESTED** [review]

**Comment by kriptonian1** [comment]
Hey @Ritika1705 and @rajdip-b using `Math.random` is not the most secure way to generate opt you can read this amazing discussion in stackexchange crypto channel to know more on why [link](https://crypto.stackexchange.com/questions/11779/why-not-just-generate-random-strings-for-one-time-password-otp). I will rather recommend to use TOTP for that you can use [otplib](https://www.npmjs.com/package/otplib). Now, using math.random is not secure because it's a pseudo random number generator, which is not cryptographically secure.

**Comment by rajdip-b** [comment]
@Ritika1705 hey! did you encounter any issue in the implementation? We are eager to merge your PR!

**Comment by Ritika1705** [comment]
Hi,
Got caught up with some work. Will try to finish it by eod. Thankyou

**Comment by rajdip-b** [comment]
Oh! Alright no issues! Whenever you are free.

File: apps/api/src/auth/service/auth.service.ts
Comment (Ritika1705) [reply apps/api/src/auth/service/auth.service.ts:1]: Hi @rajdip-b, i have removed the import.

**Comment by Ritika1705** [comment]
For better secure code generation, we can follow this article: https://medium.com/getpowerplay/a-one-time-password-otp-generator-npm-library-based-on-nanoid-79821a173798

**Comment by rajdip-b** [comment]
Hey @Ritika1705, I went through some articles, and I feel that we can use the already existing `crypto` library. Maybe you can try out this suggestion?

```ts
crypto.randomBytes(3).toString('hex'),
 ```

I'm not sure how it will perform, but let's keep things simple - and use crypto package.

**Comment by Ritika1705** [comment]
> Hey @Ritika1705, I went through some articles, and I feel that we can use the already existing `crypto` library. Maybe you can try out this suggestion?
> 
> ```ts
> crypto.randomBytes(3).toString('hex'),
> ```
> 
> I'm not sure how it will perform, but let's keep things simple - and use crypto package.

Hi @rajdip-b , I have upadated the code to use crypto package. 

The code BigInt(\0x${crypto.randomUUID().replace(/-/g, '')}`).toString().substring(0, 6)` generates a 6-character string by following these steps:

- Generate a UUID: crypto.randomUUID() generates a version 4 UUID, which looks something like 550e8400-e29b-41d4-a716-446655440000.
- Remove Hyphens: .replace(/-/g, '') removes all the hyphens from the UUID, resulting in a string like 550e8400e29b41d4a716446655440000.
- Convert to BigInt: BigInt(\0x${uuid}`)converts this hexadecimal string to a BigInt. The0x` prefix indicates that the string is in hexadecimal format.
- Convert to String: .toString() converts the BigInt to its decimal string representation.
- Extract 6 Digits: .substring(0, 6) takes the first 6 characters from the decimal string

**Review by rajdip-b — APPROVED** [review]
LGTM!

**Pull request merged**